nntrainer/app_context.cpp

```diff
@@ -404,6 +410,53 @@ const std::string AppContext::getWorkingPath(const std::string &path) {
   return getFullPath(path, working_path_base);
 }
 
+/**
+ * @brief base case of iterate_prop, iterate_prop iterates the given tuple
+ *
+ * @tparam I size of tuple(automated)
+ * @tparam V container type of properties
+ * @tparam Ts types from tuple
+ * @param prop property container to be added to
+ * @param tup tuple to be iterated
+ * @return void
+ */
+template <size_t I = 0, typename V, typename... Ts>
+typename std::enable_if<I == sizeof...(Ts), void>::type inline parse_properties(
+  V &props, std::tuple<Ts...> &tup) {
+  // end of recursion.
+}
+
+/**
+ * @brief base case of iterate_prop, iterate_prop iterates the given tuple
+ *
+ * @tparam I size of tuple(automated)
+ * @tparam V container type of properties
+ * @tparam Ts types from tuple
+ * @param prop property container to be added to
+ * @param tup tuple to be iterated
+ * @return void
+ */
+template <size_t I = 0, typename V, typename... Ts>
+  typename std::enable_if <
+  I<sizeof...(Ts), void>::type inline parse_properties(V &props,
+                                                       std::tuple<Ts...> &tup) {
+  std::string name = std::get<I>(tup);
+  std::string prop = getConfig(name);
+  if (!prop.empty())
+    props.push_back(name + "=" + prop);
+
+  parse_properties<I + 1>(props, tup);
+}
+
+const std::vector<std::string> AppContext::getProperties(void) {
+  std::vector<std::string> properties;
+
+  auto props = std::tuple("memory_swap", "memory_swap_path");
```

Comment:
How about merge the ```layer``` property in here? then we could remove 172 line.

Reply:
`layer` is special property which is only used in app_context. And the behaviour is different.
It only add `paths` to pluginPath while 'getProperties' adds `prop=value` to propergate to network property.